widgetastic_manageiq/__init__.py

```diff
@@ -3576,7 +3576,7 @@ def _current_page_elements(self):
         if self.browser.product_version < '5.9':
             br = self.browser
             for el in br.elements(self.elements):
-                el_id = int(br.get_attribute('href', el).split('/')[-1])
+                el_id = br.get_attribute('href', el).split('/')[-1]
```

Comment:
**QUESTION**
Why is it considered as a string?

Reply:
because all entities have string id in non 0 regions. It looks like 1r0001

Reply:
@quarckster: 5.8's URL pattern have change in for quadicon links, `?link=true` at last part, check this (**Reference:** https://github.com/ManageIQ/integration_tests/commit/73f4b629884265d36c4bae5c5879fd75aa4d41b9)

![screenshot from 2018-02-16 14-23-21](https://user-images.githubusercontent.com/10824880/36299895-09688320-1325-11e8-8a1a-b0a296f4229f.png)

Reply:
hm, but ?link isn't part of id. everything after id has to be removed then

Reply:
If we only wanted id(number) there, we have to change
`el_id = int(br.get_attribute('href', el).split('/')[-1]).split('?')[0])`
Note `.split('?')[0])` there. but then it fails in `if item(entity_id=self.entity_id).is_displayed:` 

This is where it should failed
```
class NonJSBaseEntity(View):
    quad_entity = BaseQuadIconEntity
  
    def __init__(self, parent, entity_id, name=None, logger=None):
        View.__init__(self, parent, logger=logger)
        self.entity_id = entity_id
        self._name = name

    def _get_existing_entity(self):
        for item in (self.quad_entity, self.tile_entity, self.list_entity):
        >>  if item(entity_id=self.entity_id).is_displayed:
```

This is b'cause `ROOT` of `BaseQuadIconEntity` need to changed so that it should parse `?link=true` from `href` and if you change `ROOT` to new one i.e.
```
    ROOT = ParametrizedLocator('.//table[./tbody/tr/td/*[(self::a or self::span) and '
                               'substring(@href, string-length(@href)'
                               '-string-length("/{entity_id}")+1)="/{entity_id}"]]' or
                           >    './/table[./tbody/tr/td/*[(self::a or self::span) and '
                           >    'substring(@href, string-length(@href)'
                           >   '-string-length("/{entity_id}?link=true")+1)="/{entity_id}?link=true"]]')
```
This all workout okay (I have done this before)! but I personally feel removal of `int()` if better approach than things mentioned above as I see https://github.com/ManageIQ/integration_tests/commit/73f4b629884265d36c4bae5c5879fd75aa4d41b9 what say @quarckster @izapolsk?

Reply:
I'm looking into this issue and will add my commit with fix in a bit. Please update PR according to @quarckster 's remarks.

Reply:
@Yadnyawalkya, please use  there
el_id = br.get_attribute('href', el).split('?')[0].split('/')[-1]